amlip_cpp/src/cpp/types/AmlipIdDataType.cpp

```diff
@@ -0,0 +1,305 @@
+std::string AmlipIdDataType::name() const
+{
+    return std::string(std::begin(name_), std::end(name_));
+}
+
+void AmlipIdDataType::name(const std::array<uint8_t, NAME_SIZE>& name)
+{
+    name_ = name;
+}
+
+const std::array<uint8_t, NAME_SIZE>& AmlipIdDataType::base64_name() const
+{
+    return name_;
+}
+
+const std::array<uint8_t, RAND_SIZE>& AmlipIdDataType::id() const
+{
+    return rand_id_;
+}
+
+void AmlipIdDataType::id(const std::array<uint8_t, RAND_SIZE>& id)
+{
+    rand_id_ = id;
+}
+
+const char* AmlipIdDataType::type_name()
+{
+    return "AMLIP-ID";
+}
+
+bool AmlipIdDataType::is_defined() const noexcept
+{
+    return (*this) != UNDEFINED_ID_;
+}
+
+AmlipIdDataType AmlipIdDataType::new_unique_id()
+{
+    // initialize the random number generator
+    srand (time(NULL));
+
+    std::array<uint8_t, NAME_SIZE> name;
+    for (int i=0; i<NAME_SIZE; i++)
+    {
+        // generate a random char between 'a' and 'z'
+        char c = 'a' + static_cast<char>(rand() % 26);
+        name[i] = static_cast<uint8_t>(c);
+    }
+
+    std::array<uint8_t, RAND_SIZE> rand_id;
+    for (int i=0; i<RAND_SIZE; i++)
+    {
+        // generate a random number between 0 and 255
+        uint8_t num = static_cast<uint8_t>(rand() % 256);
+        rand_id[i] = num;
+    }
+
+    return AmlipIdDataType(name, rand_id);
+}
+
+AmlipIdDataType AmlipIdDataType::new_unique_id(const std::string& name)
+{
+    AmlipIdDataType new_id = new_unique_id();
+    new_id.name(str_name_to_array(name));
+
+    return new_id;
+}
+
+AmlipIdDataType AmlipIdDataType::new_unique_id(const char* name)
+{
+    return new_unique_id(std::string(name));
+}
+
+AmlipIdDataType AmlipIdDataType::undefined_id()
+{
+    return UNDEFINED_ID_;
+}
+
+std::array<uint8_t, NAME_SIZE> AmlipIdDataType::str_name_to_array(const std::string& name)
+{
+    std::array<uint8_t, NAME_SIZE> _name;
+    std::string name_substr = name.substr(0, NAME_SIZE);
+    std::copy(name_substr.begin(), name_substr.end(), _name.data());
+
+    return _name;
+}
+
+void AmlipIdDataType::serialize(
+    eprosima::fastcdr::Cdr& scdr) const
+{
+    scdr << name_;
+    scdr << rand_id_;
+}
+
+void AmlipIdDataType::deserialize(
+    eprosima::fastcdr::Cdr& dcdr)
+{
+    dcdr >> name_;
+    dcdr >> rand_id_;
+}
+
+void AmlipIdDataType::serialize_key(
+    eprosima::fastcdr::Cdr& scdr) const
+{
+    (void) scdr;
+}
+
+size_t AmlipIdDataType::get_max_cdr_serialize_size(
+    size_t current_alignment)
+{
+    size_t initial_alignment = current_alignment;
+    current_alignment += ((NAME_SIZE) * 1) + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+    current_alignment += ((RAND_SIZE) * 1) + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+
+    return current_alignment - initial_alignment;
+}
+
+size_t AmlipIdDataType::get_cdr_serialize_size(
+    const AmlipIdDataType& data,
+    size_t current_alignment)
+{
+    (void)data;
+    size_t initial_alignment = current_alignment;
+
+    if ((NAME_SIZE) > 0)
```

Comment:
NIT:
In our case, as we now these values are bigger than 0, maybe we can skip this.
This will be useful in case we do a General (template) class for any ID, where these values are parameters.